messenger/make.py

```diff
@@ -1,112 +1,138 @@
+    cfg.read(config)
+
+    return dict(cfg.items(host))
 
 
 def do_build(make_cmd, messenger_exe):
     print('Building %s...' % messenger_exe)
     print(make_cmd)
     messenger_dir = get_messenger_dir()
-    subprocess.check_output(shlex.split(make_cmd), shell=True)
+    subprocess.call(shlex.split(make_cmd), stderr=subprocess.STDOUT)
 
     messenger_loc = os.path.join(messenger_dir, messenger_exe)
 
-    shutil.move(messenger_exe, messenger_loc)
+    shutil.move(messenger_exe, os.path.join('messenger', messenger_loc))
 
-    if os.path.exists('messenger.o'):
+    if os.path.isfile('messenger.o'):
         os.remove('messenger.o')
 
 
-def build_octave():
+def build_octave(static=False):
     paths = "-L%(octave_lib)s -I%(octave_inc)s -L%(zmq_lib)s -I%(zmq_inc)s"
-    paths = paths % get_config()
+    paths = paths % get_config(platform.system())
     make_cmd = "mkoctfile --mex %s -lzmq ./src/messenger.c" % paths
     do_build(make_cmd, 'messenger.mex')
 
 
 def build_matlab(static=False):
-    """build the messenger mex for MATLAB
-    
+    """
+    Build the messenger mex for MATLAB
+
+    Parameters
+    ----------
     static : bool
         Determines if the zmq library has been statically linked.
         If so, it will append the command line option -DZMQ_STATIC
         when compiling the mex so it matches libzmq.
     """
-    cfg = get_config()
-    # To deal with spaces, remove quotes now, and add
-    # to the full commands themselves.
-    matlab_bin = cfg['matlab_bin'].strip('"')
-    # Get the extension
-    extcmd = '"' + os.path.join(matlab_bin, "mexext") + '"'
-    extension = subprocess.check_output(extcmd, shell=True)
-    extension = extension.decode('utf-8').rstrip('\r\n')
+    matlab_bin, cfg = settings.get_matlab_bin(), get_config(platform.system())
+
+    extcmd     = '%s' % os.path.join(matlab_bin, "mexext")
+    extension  = subprocess.check_output(extcmd, shell=True)
+    extension  = extension.decode('utf-8').rstrip()
 
     # Build the mex file
     mex = '"' + os.path.join(matlab_bin, "mex") + '"'
-    paths = "-L%(zmq_lib)s -I%(zmq_inc)s" % cfg
-    make_cmd = '%s -O %s -lzmq ./src/messenger.c' % (mex, paths)
+    paths    = "-L'%(zmq_lib)s' -I'%(zmq_inc)s'" % cfg
+    make_cmd = '%s -O %s -lzmq messenger/src/messenger.c' % (mex, paths)
+
     if static:
         make_cmd += ' -DZMQ_STATIC'
+
     do_build(make_cmd, 'messenger.%s' % extension)
 
 
-if __name__ == '__main__':
-    import argparse
-    parser = argparse.ArgumentParser()
-    parser.add_argument(
-        "target",
-        choices=["matlab", "octave"],
-        type=str.lower,
-        help="target to be built")
-    parser.add_argument("--static", action="store_true",
-                        help="staticly link libzmq")
-    args = parser.parse_args()
-    if args.target == "matlab":
-        build_matlab(static=args.static)
-    elif args.target == "octave":
-        build_octave()
+def get_matlab_bin(config='config.ini'):
```

Comment:
This function is never called.It looks like settings.get_matlab_bin is called instead, but the script in settings.py is actually called settings.get_matlab_env